tests/shardddl3/run.sh

```diff
@@ -873,6 +873,1442 @@ function DM_112() {
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (a int primary key, b int);\"" \
+        "clean_table" "pessimistic"
+    run_sql_tidb "create database if not exists ${shardddl}; create table ${shardddl}.${tb} (a int, b int, primary key(a) nonclustered);"
+    run_case 132 "double-source-optimistic" \
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, b int);\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, b int);\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (a int primary key, b int);\"" \
+        "clean_table" "optimistic"
+}
+
+function DM_133_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} drop primary key, add primary key(a);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb1} drop primary key, add primary key(a);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb2} drop primary key, add primary key(a);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(10,10);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(11,11);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(12,12);"
+
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+}
+
+# Shrink the primary key field.
+function DM_133 {
+    run_sql_tidb "create database if not exists ${shardddl}; create table ${shardddl}.${tb} (a int, b int, primary key(a,b) nonclustered);"
+    run_case 133 "double-source-pessimistic" \
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b int, primary key(a,b));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b int, primary key(a,b));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b int, primary key(a,b));\"" \
+        "clean_table" "pessimistic"
+    run_sql_tidb "create database if not exists ${shardddl}; create table ${shardddl}.${tb} (a int, b int, primary key(a,b) nonclustered);"
+    run_case 133 "double-source-optimistic" \
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b int, primary key(a,b));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b int, primary key(a,b));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b int, primary key(a,b));\"" \
+        "clean_table" "optimistic"
+}
+
+function DM_134_CASE {
```

Comment:
this case is marked as negative in spreadsheet

Reply:
Yes, dm should report a warning here. I will add a FIXME/TODO here.